library/psa_crypto.c

```diff
@@ -6480,6 +6480,47 @@ static psa_status_t psa_key_derivation_input_internal(
     return status;
 }
 
+static psa_status_t psa_key_derivation_input_integer_internal(
```

Comment:
@silabs-hannes I have added another commit in this PR which includes the struct declarations. Should anything more be added?

Reply:
I should have been more precise in my last comment, but my actual ask was just to add the PBKDF2 logic to this function. And in order to do that correctly, we at least need some minimal definition of the PBKDF2 operation struct containing a cost (iteration count) integer. It also looks to me like we should add support for PBKDF2 in the setup function, otherwise we should not be allowed to set the cost in this function (which I interpret as the purpose of the task we're working on here).

This is not explicitly mentioned by the task description, so it's possible that it should be raised as a prerequisite PR before this one. I'll leave that up to your judgement if you think that the PR gets to big (can always check with Manuel as well).